test_server.py

```diff
@@ -43,36 +70,40 @@ def test_get_product(self):
 
     def test_query_product_list(self):
         self.setup_test_by_attribute('category', 'entertainment')
-        self.setup_test_by_attribute('discontinued', True)    
+        self.setup_test_by_attribute('discontinued', True)  
+        self.setup_test_by_attribute('price', '999')  
 
     def test_create_product(self):
         # save the current number of products for later comparison
         product_count = self.get_product_count()
         # add a new product
-        new_product = {'name': 'toaster', 'category': 'kitchen appliances'}
+        new_product = {'name': 'toaster', 'category': 'kitchen appliances', 'price': '99'}
         data = json.dumps(new_product)
         resp = self.app.post('/products', data=data, content_type='application/json')
         self.assertTrue( resp.status_code == HTTP_201_CREATED )
         new_json = json.loads(resp.data)
         self.assertTrue (new_json['name'] == 'toaster')
-        # Check default value of 'discontinued' is 'False'
+        # check default value of 'discontinued' is 'False'
         self.assertTrue (new_json['discontinued'] == False)
+        # check the price of the product
+        self.assertTrue (new_json['price'] == '99')
         # check that count has gone up and includes sammy
-        resp = self.app.get('/products')
+        resp = self.app.get('/products?limit=' + str(product_count + 1))
```

Comment:
I think this is still necessary. If I change the default cutoff to 10 and we choose to add more products in our test_server (which I think we should do), then if the number of products we have in the test is already 10, line 91 will produce an error without that 'limit'.

Does that make sense?

I can choose to drop it once if I increase the cutoff to 10, GIVEN that the number of products in our test_server will always remain less than 10. I have this in just in case in the future we choose to add more products to our test cases.